esp/platform/espp.hpp

```diff
@@ -232,6 +232,47 @@ class CEspServer : public CInterface,
         }
     }
 
+    virtual void removeBinding(unsigned short port, IEspRpcBinding & bind)
+    {
+        IEspProtocol* prot = dynamic_cast<IEspProtocol*>(bind.queryListener());
+        if (prot)
+        {
+            int left = prot->removeBindingMap(port, &bind);
+            if (left == 0)
+            {
+                DBGLOG("No more bindings on port %d, so freeing up the port.",port);
+                ISocket **socketp = m_srvSockets.getValue(port);
```

Comment:
Is there a potential race with addBinding?  E.g., if an add occurs at the same time as the last remove the binding could be removed, a new one added, then the socket removed - so nothing will be listening to the port.